File: C++/Rectangle/Rectangle.cpp
#include "Rectangle.h"

int Rectangle::numOfRect = 0;

Rectangle::Rectangle()
{
    numOfRect++;
}
Rectangle::Rectangle(double newhight, double newwidth)
{
    height = newhight;
    width = newwidth;
    numOfRect++;
}
void Rectangle::setHeight(double newhight)
{
    height = newhight;
}
void Rectangle::setWidth(double newwidth)
{
    width = newwidth;
}
double Rectangle::getHeight() const
{
    return height;
}
double Rectangle::getWidth() const
{
    return width;
}
double Rectangle::getArea() const
{
    return height * width;
}
int Rectangle::getNumOfRect()
{
    return numOfRect;
}

File: Python/Rectangle.py
from GeometricObject import GeometricObject

class Rectangle(GeometricObject):
    def __init__(self,height,width,color,filled):
        super().__init__(color,filled)
        self.__height=height
        self.__width=width
        
    def getWidth(self):
        return self.__width
        
    def getHeight(self):
        return self.__height
        
    def setWidth(self,width):
        self.__width=width
        
    def setHeight(self,height):
        self.__height=height
        
    def getArea(self):
        return self.__width*self.__height
        
    def getPerimeter(self):
        return (self.__width+self.__height)*2